libcore/ListModel.vala

```diff
@@ -315,15 +316,28 @@ public class Files.ListModel : Gtk.TreeStore, Gtk.TreeModel {
             var dir = Files.Directory.from_file (file);
             dir.cancel ();
             Gtk.TreeIter? child_iter = null;
+            Files.File child_file;
             // Remove all child nodes so they are refreshed if subdirectory reloaded
             // Faster than checking for duplicates
+            set_sorting_off ();
             if (iter_children (out child_iter, parent_iter)) {
-                while (remove (ref child_iter)) {};
+                get (child_iter, ColumnID.FILE_COLUMN, out child_file);
+                if (child_file != null) {
+                    file_treerow_map.unset (child_file.uri);
+                }
+
+                while (remove (ref child_iter)) {
+                    get (child_iter, ColumnID.FILE_COLUMN, out child_file);
+                    if (child_file != null) {
+                        file_treerow_map.unset (child_file.uri);
+                    }
+                };
```

Comment:
It would be even better if we can write code which don't have the following lines twice… :thinking: 

```vala
get (child_iter, ColumnID.FILE_COLUMN, out child_file);
if (child_file != null) {
    file_treerow_map.unset (child_file.uri);
}
```